test/common/network/dns_impl_test.cc

```diff
@@ -111,8 +111,11 @@ class TestDnsServerQuery {
 
         // Expect requests to be small, so stack allocation is fine for test code.
         unsigned char* request = static_cast<unsigned char*>(buffer_.linearize(size_));
-        // Only expecting a single question.
+// Only expecting a single question.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
```

Comment:
Yeah, this seems pretty ugly. Can we fix the gRPC c-ares BUILD to do something similar to what we did in https://github.com/envoyproxy/envoy/blob/master/ci/prebuilt/BUILD#L9? This handles the includes as system includes IIRC, which avoids Wold-style-cast.